### PR TITLE
Fixed missing points in scatter series with boost

### LINKED ISSUE #21897: Scatter series and boost + multiple axes missing points
#### Expected behaviour
all of the points on the plot render correctly. this worked in 11.1.0 but does not work in 11.4.7. 

#### Actual behaviour
when the plot is boosted due to many series being present the symbols do not render but they are hit testable so the kd-tree is correct. I can only assume that the point feel they are clipped due to the scale range/positioning

#### Live demo with steps to reproduce
current version: https://jsfiddle.net/7fc8n46L/1/
11.1.0 version: https://jsfiddle.net/wfkrd5zq/

#### Product version
11.4.7

#### Affected browser(s)
chrome latest and safari

### PR DESCRIPTION
**Pull request by Eskils**
Fixed #21897, missing points in scatter series with boost

This fix changes the behavior for scatter and bubble series (where sorted is false).  The optimization attempts made in `pushSeriesData` does not work for these series and so are bypassed. 

Two new samples are added to test and make sure no regressions occur:
- samples/highcharts/boost/scatter-points-outside-zoom-range-visible-19701
- samples/highcharts/boost/scatter-missing-points-21897

---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1208406417591641

### CODE REVIEW

**Review by bre1470 — CHANGES_REQUESTED** [review]
Fix seems to work! 👍
The visual tests should be merged into `samples/highcharts/boost/scatter/` to save overhead, just expand there the setup to cover you second test. `samples/issues` is legacy.

**Comment by Eskils** [comment]
Thank you for pointing this out. 

I moved the samples to `samples/highcharts/boost`. Not sure what you mean with expanding the tests in `samples/highcharts/boost/scatter`—can there be multiple visual tests in one sample? 

Maybe I misunderstand?

**Review by goransle — APPROVED** [review]
I agree with the points above☝️, but otherwise LGTM

**Comment by Eskils** [comment]
I converted the visual test into a unit test.

**Review by bre1470 — APPROVED** [review]
Sorry for the delay. Looks good! 👍

File: ts/Extensions/Boost/WGLRenderer.ts [hunk 1]
```diff
@@ -881,10 +889,15 @@ class WGLRenderer {
 
             // The first point before and first after extremes should be
             // rendered (#9962, 19701)
+            // Boost fails to render scatter series with a single point (#21897)
```
Comment (TorsteinHonsi) [inline ts/Extensions/Boost/WGLRenderer.ts:892]: "Boost fails..." describes a situation that is no longer true, so this comment would be confusing to the future us. This is more precise:
```suggestion
            // Make sure series with a single point are rendered (#21897)
```

File: ts/Extensions/Boost/WGLRenderer.ts [hunk 2]
```diff
@@ -859,6 +862,11 @@ class WGLRenderer {
                 isYInside = y >= yMin && y <= yMax;
             }
 
+            // Scatter points outside zoomed range were visible (#19701)
```
Comment (TorsteinHonsi) [inline ts/Extensions/Boost/WGLRenderer.ts:865]: Describe how the code actually works, not how it used to fail at some point:
```suggestion
            // Do not render points outside the zoomed range (#19701)
```

**Review by TorsteinHonsi — CHANGES_REQUESTED** [review]
Thanks, nice work! Just nitpicking on the language of the comments.

**Comment by Eskils** [comment]
Thank you for your insight. I will change my other PRs if I have made the same mistake there. 😅

**Review by TorsteinHonsi — APPROVED** [review]

**Pull request merged**